test/extended/networking/services.go

```diff
@@ -247,4 +248,83 @@ var _ = Describe("[sig-network] services", func() {
 			}
 		})
 	})
+
+	InKubeVirtClusterContext(oc, func() {
+
+		mgmtFramework := e2e.NewDefaultFramework("mgmt-framework")
+		mgmtFramework.SkipNamespaceCreation = true
+
+		f1 := e2e.NewDefaultFramework("server-framework")
+		f1.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
+
+		It("should allow connections to pods from infra cluster pod via NodePort across different infra nodes", func() {
+
+			// This tests connectivity from the infra cluster's pod network to a NodePort
+			// within a nested KubeVirt Hypershift guest cluster.
+			//
+			// This exercises the back half of the network flow used to pass ingress
+			// through the ingress cluster into the nested guest cluster.
+			//
+			// client pod (on infra cluster) -> guest node IP -> NodePort -> server pod (on guest cluster)
+			//
+			_, hcpNamespace, err := exutil.GetHypershiftManagementClusterConfigAndNamespace()
+			Expect(err).To(Succeed())
+
+			oc = exutil.NewHypershiftManagementCLI(hcpNamespace).AsAdmin()
+
+			mgmtClientSet := oc.KubeClient()
+			mgmtFramework.Namespace = &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: hcpNamespace,
+				},
+			}
+			mgmtFramework.ClientSet = mgmtClientSet
+
+			Expect(checkKubeVirtInfraClusterNodePortConnectivity(f1, mgmtFramework, oc)).To(Succeed())
+		})
+
+		It("should allow connections to pods from guest hostNetwork pod via NodePort across different guest nodes", func() {
+			// Within a nested KubeVirt guest cluster, this tests the ability for
+			// NodePort services to route from hostnet to pods across guest nodes.
+			Expect(checkKubeVirtGuestClusterNodePortConnectivity(f1, f1)).To(Succeed())
+		})
+
+		It("should allow direct connections to pods from guest cluster pod in pod network across different guest nodes", func() {
+			// Within a nested KubeVirt guest cluster, this tests the ability for different pods within the
+			// guest cluster to communicate each other, across different guest cluster nodes, via PodNetwork.
+			Expect(checkKubeVirtGuestClusterPodNetworkConnectivity(f1, f1)).To(Succeed())
```

Comment:
does the generic `should allow connections to another pod on a different node via a service IP`  test already cover this?

Reply:
Not exactly. In this test case we're using the target pod's internal IP, without exposing it as a service in the cluster.
I agree it should behave the same, since the ServiceIP is being translated to the Pod IP.
If using the direct IP of the target pod doesn't add any value, we can drop it. What do you think?

Reply:
ah, alrighty, yeah lets leave the test then